### PR TITLE
11.0 mig pos payment terminal

### PR DESCRIPTION
**Pull request by cvinh**

### CODE REVIEW

File: pos_payment_terminal/README.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   :alt: License: AGPL-3
+
+====================
+POS Payment Terminal
+====================
+
+This module adds support for credit card reader and checks printer
+in the Point of Sale.
+
+This module has been developped during a POS code sprint at Akretion
```
Comment (tbaden) [inline pos_payment_terminal/README.rst:12]: developed (typo)

File: pos_payment_terminal/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'POS Payment Terminal',
+    'version': '11.0.0.1.1',
+    'category': 'Point Of Sale',
+    'summary': 'Manage Payment Terminal device from POS front end',
+    'author': "Aurélien DUMAINE,Akretion,Odoo Community Association (OCA)",
+    'license': 'AGPL-3',
+    'depends': ['point_of_sale'],
```
Comment (tbaden) [inline pos_payment_terminal/__manifest__.py:13]: please add website tag with value: https://github.com/OCA/pos

File: pos_payment_terminal/static/src/js/pos_payment_terminal.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                if (lines[i].cid === line_cid) {
+                    line = lines[i];
+                }
+            }
+
+            var data = {'amount' : line.get_amount(),
+                        'currency_iso' : currency_iso,
+                        'currency_decimals' : currency_decimals,
+                        'payment_mode' : line.cashregister.journal.payment_mode};
+            //console.log(JSON.stringify(data));
```
Comment (tbaden) [inline pos_payment_terminal/static/src/js/pos_payment_terminal.js:36]: remove commented log

File: pos_payment_terminal/static/src/js/pos_payment_terminal.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    });
+
+
+    screens.PaymentScreenWidget.include({
+        render_paymentlines : function(){
+        this._super.apply(this, arguments);
+            var self  = this;
+            this.$('.paymentlines-container').unbind('click').on('click', '.payment-terminal-transaction-start', function(event){
+            // Why this "on" thing links severaltime the button to the action if I don't use "unlink" to reset the button links before ?
+            //console.log(event.target);
```
Comment (tbaden) [inline pos_payment_terminal/static/src/js/pos_payment_terminal.js:48]: remove commented log

File: pos_payment_terminal/views/pos_config.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_pos_config_form" model="ir.ui.view">
```
Comment (tbaden) [inline pos_payment_terminal/views/pos_config.xml:4]: use original id in inherit (optional)

**Review by tbaden — CHANGES_REQUESTED** [review]
thanks for the work 👍 
i did a code review. just some minor changes

File: pos_payment_terminal/views/account_journal.xml [hunk 6]
```diff
@@ -6,9 +6,9 @@
         <field name="model">account.journal</field>
         <field name="inherit_id" ref="point_of_sale.view_account_journal_pos_user_form"/>
         <field name="arch" type="xml">
-            <field name="amount_authorized_diff" position="after">
+            <xpath expr="//field[@name='journal_user']" position="after">
```
Comment (chienandalu) [inline pos_payment_terminal/views/account_journal.xml:9]: Why do you change the location of the setting?

File: pos_payment_terminal/__manifest__.py [hunk 7]
```diff
@@ -5,7 +5,7 @@
 
 {
     'name': 'POS Payment Terminal',
-    'version': '10.0.0.1.1',
+    'version': '11.0.0.1.1',
```
Comment (chienandalu) [inline pos_payment_terminal/__manifest__.py:8]: It should be `11.0.1.0.0`

File: pos_payment_terminal/README.rst [hunk 8]
```diff
@@ -0,0 +1,74 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (chienandalu) [inline pos_payment_terminal/README.rst:1]: Please, try to use OCA readme fragments https://github.com/OCA/maintainer-tools/tree/master/template/module/readme

File: pos_payment_terminal/__manifest__.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
```
Comment (chienandalu) [inline pos_payment_terminal/__manifest__.py:1]: No utf8 shebang needed in python3 (>=v11)

File: pos_payment_terminal/models/account_journal.py [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import models, fields
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    payment_mode = fields.Selection(
```
Comment (chienandalu) [inline pos_payment_terminal/models/account_journal.py:13]: Maybe a more specific field name should be used (something like `terminal_payment_mode`) as this is a model used for other workflows.

File: pos_payment_terminal/static/src/js/pos_payment_terminal.js (hunk 4 above)
Comment (chienandalu) [inline pos_payment_terminal/static/src/js/pos_payment_terminal.js:48]: remove log debugging comments

File: pos_payment_terminal/static/src/js/pos_payment_terminal.js [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+odoo.define('pos_payment_terminal.pos_payment_terminal', function (require) {
+    "use strict";
+
+    var screens = require('point_of_sale.screens');
+    var devices = require('point_of_sale.devices');
+    var models = require('point_of_sale.models');
+    var core = require('web.core');
+    var _t = core._t;
+    var QWeb = core.qweb;
```
Comment (chienandalu) [inline pos_payment_terminal/static/src/js/pos_payment_terminal.js:18]: remove declarations if not needed

**Review by chienandalu — CHANGES_REQUESTED** [review]
Thanks for the work! Some comments:

**Comment by cvinh** [comment]
I'm working on it

File: pos_payment_terminal/models/account_journal.py (hunk 10 above)
Comment (rousseldenis) [reply pos_payment_terminal/models/account_journal.py:13]: Or maybe migrating to payment.mode model as in https://github.com/OCA/bank-payment/tree/10.0/account_payment_mode which could be more correct

**Comment by ivantodorovich** [comment]
Hi @cvinh,

First of all, thanks so much for the work done!

Are you still working on this? We'd like to pick up where you left, if you aren't. 
Can you confirm?

**Comment by legalsylvain** [comment]
closing, due to inactivity. Feel free to reopen.

**Pull request closed without merging**